Source/WebCore/rendering/style/BorderData.h

```diff
@@ -107,68 +108,61 @@ class BorderData {
     };
 
     template<BoxSide side> struct ColorAccessor {
-        static const Style::Color& get(const BorderData& data) { return data.edges()[side].m_color; }
-        static void set(BorderData& data, Style::Color&& color) { data.edges()[side].m_color = WTFMove(color); }
+        static const Style::Color& get(const BorderData& data) { return data.edges[side].color; }
+        static void set(BorderData& data, Style::Color&& color) { data.edges[side].color = WTFMove(color); }
     };
     using BorderColorsView = BorderEdgesView<false, ColorAccessor, const Style::Color&, Style::Color&&>;
     using BorderColorsConstView = BorderEdgesView<true, ColorAccessor, const Style::Color&, Style::Color&&>;
     BorderColorsView colors() { return { .borderData = *this }; }
     BorderColorsConstView colors() const { return { .borderData = *this }; }
 
     template<BoxSide side> struct StyleAccessor {
-        static unsigned get(const BorderData& data) { return data.edges()[side].m_style; }
-        static void set(BorderData& data, unsigned style) { data.edges()[side].m_style = style; }
+        static unsigned get(const BorderData& data) { return data.edges[side].style; }
+        static void set(BorderData& data, unsigned style) { data.edges[side].style = style; }
     };
     using BorderStylesView = BorderEdgesView<false, StyleAccessor, unsigned>;
     using BorderStylesConstView = BorderEdgesView<true, StyleAccessor, unsigned>;
     BorderStylesView styles() { return { .borderData = *this }; }
     BorderStylesConstView styles() const { return { .borderData = *this }; }
 
-    RectEdges<BorderValue>& edges() { return m_edges; }
-    const RectEdges<BorderValue>& edges() const { return m_edges; }
-
-    BorderValue& left() { return m_edges.left(); }
-    BorderValue& right() { return m_edges.right(); }
-    BorderValue& top() { return m_edges.top(); }
-    BorderValue& bottom() { return m_edges.bottom(); }
-
-    const BorderValue& left() const { return m_edges.left(); }
-    const BorderValue& right() const { return m_edges.right(); }
-    const BorderValue& top() const { return m_edges.top(); }
-    const BorderValue& bottom() const { return m_edges.bottom(); }
-
-    Style::BorderImage& image() { return m_image; }
-    const Style::BorderImage& image() const { return m_image; }
+    BorderValue& left() { return edges.left(); }
+    BorderValue& right() { return edges.right(); }
+    BorderValue& top() { return edges.top(); }
+    BorderValue& bottom() { return edges.bottom(); }
```

Comment:
Do all these functions that return references need `LIFETIME_BOUND` macros?

Reply:
Yeah, as do all the others in that file. 

Can’t wait for the new clang option to recommend missing lifetime attributes!